pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -2328,6 +2334,31 @@ func (dm *DockerManager) SyncPod(pod *v1.Pod, _ v1.PodStatus, podStatus *kubecon
 	return
 }
 
+// ensureDirectoryAccess ensures that if we are in a remap environment and the pod has FSGroup set that
```

Comment:
at this point nothing would be surfaced.  IMO the implementations of the runtimes are different enough to want to hide this logic and act on a flag to tell the implementation that we want user ns behavior whatever that may be.

In the future I think taking a remap uid/gid as an argument and letting the orchestration figure out what those values are is the way to go.  Then the kubelet would know what to expect up front and could adjust accordingly.  

We could allow the kubelet to call a method on the CRI but if there are many steps that the CRI needs to take (due to differing impls) then I think we pollute the CRI or end up trying to manage lifecycle methods differently.